lib/charms/vault_k8s/v0/vault_autounseal.py

```diff
@@ -0,0 +1,518 @@
+        vault_address: str,
+        mount_path: str,
+        key_name: str,
+        approle_role_id: str,
+        approle_secret_id: str,
+        ca_certificate: str,
+    ) -> None:
+        """Set the autounseal data in the relation databag.
+
+        Args:
+            relation: The Juju relation to set the autounseal data in.
+            vault_address: The address of the Vault server which will be used for autounseal
+            mount_path: The path to the transit engine mount point where the key is stored.
+            key_name: The name of the transit key to use for autounseal.
+            approle_role_id: The AppRole Role ID to use when authenticating with the external Vault server.
+            approle_secret_id: The AppRole Secret ID to use when authenticating with the external Vault server.
+            ca_certificate: The CA certificate to use when validating the external Vault server's certificate.
+        """
+        if not self.charm.unit.is_leader():
+            logger.warning(
+                "Attempting to set the auto-unseal data without being the leader. Ignoring the request."
+            )
+            return
+        if relation is None:
+            logger.warning("No relation found")
+            return
+        if not relation.active:
+            logger.warning("Relation is not active")
+            return
+        credentials_secret_id = self._create_autounseal_credentials_secret(
+            relation, approle_role_id, approle_secret_id
+        )
+        relation.data[self.charm.app].update(
+            {
+                "address": vault_address,
+                "mount_path": mount_path,
+                "key_name": key_name,
+                "credentials_secret_id": credentials_secret_id,
+                "ca_certificate": ca_certificate,
+            }
+        )
+
+    def get_outstanding_requests(self, relation_id: int | None = None) -> List[Relation]:
+        """Get the outstanding requests for the relation.
+
+        This will retrieve any vault-autounseal relations that have not yet had
+        credentials issued for them.
+        """
+        outstanding_requests: List[Relation] = []
+        requirer_requests = self.get_active_relations(relation_id=relation_id)
+        for relation in requirer_requests:
+            if not self._credentials_issued_for_request(relation_id=relation.id):
+                outstanding_requests.append(relation)
+        return outstanding_requests
+
+    def get_active_relations(self, relation_id: int | None = None) -> List[Relation]:
+        """Get all active relations on the relation name this class was initialized with.
+
+        Args:
+            relation_id: The relation ID to filter by. If None, all active relations are returned.
+
+        Returns:
+            A list of active relations.
+        """
+        relations = (
+            [
+                relation
+                for relation in self.model.relations[self.relation_name]
```

Comment:
Not sure I get this. What's the difference between `self.model.relations[self.relation_name]` and `self.model.relations.get(self.relation_name)`?

Reply:
I don't think there is any difference, it's a mapping and both the operator and the `get` function should work.
I fixed it because this can cause a `keyError`